tst/com/amazon/corretto/crypto/provider/test/AesXtsTest.java

```diff
@@ -0,0 +1,605 @@
+    } catch (final NoSuchAlgorithmException e) {
+      throw new RuntimeException(e);
+    } catch (NoSuchPaddingException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static class AesXtsTestCase {
+    public String keyHex;
+    public String tweakHex;
+    public String inputHex;
+    public String cipherTextHex;
+
+    AesXtsTestCase() {
+      keyHex = "";
+      tweakHex = "";
+      inputHex = "";
+      cipherTextHex = "";
+    }
+
+    AesXtsTestCase addKey(final String data) {
+      keyHex += data;
+      return this;
+    }
+
+    AesXtsTestCase addTweak(final String data) {
+      tweakHex += data;
+      return this;
+    }
+
+    AesXtsTestCase addInput(final String data) {
+      inputHex += data;
+      return this;
+    }
+
+    AesXtsTestCase addCipherText(final String data) {
+      cipherTextHex += data;
+      return this;
+    }
+
+    void checkPositive() {
+      try {
+        final SecretKeySpec key = new SecretKeySpec(TestUtil.decodeHex(keyHex), "AES-XTS");
+        final byte[] input = TestUtil.decodeHex(inputHex);
+        final IvParameterSpec tweak = new IvParameterSpec(TestUtil.decodeHex(tweakHex));
+
+        final Cipher cipher = AesXtsTest.getAesXtsCipher();
+        cipher.init(Cipher.ENCRYPT_MODE, key, tweak);
+        final byte[] actualCipherText = cipher.doFinal(input);
+        assertArrayEquals(TestUtil.decodeHex(cipherTextHex), actualCipherText);
+
+        cipher.init(Cipher.DECRYPT_MODE, key, tweak);
+        assertArrayEquals(input, cipher.doFinal(actualCipherText));
+      } catch (final Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    void checkNegative() {
+      // The last byte of expected cipher text is expected to be tampered.
+      try {
+        final SecretKeySpec key = new SecretKeySpec(TestUtil.decodeHex(keyHex), "AES-XTS");
+        final byte[] input = TestUtil.decodeHex(inputHex);
+        final IvParameterSpec tweak = new IvParameterSpec(TestUtil.decodeHex(tweakHex));
+
+        final Cipher cipher = AesXtsTest.getAesXtsCipher();
+        cipher.init(Cipher.ENCRYPT_MODE, key, tweak);
+        final byte[] actualCipherText = cipher.doFinal(input);
+        final byte[] expectedCipherText = TestUtil.decodeHex(cipherTextHex);
+        // The last byte of expected cipher text is expected to be tampered.
+        assertNotEquals(lastElement(expectedCipherText), lastElement(actualCipherText));
+      } catch (final Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    private static byte lastElement(final byte[] data) {
+      return data[data.length - 1];
+    }
+  }
+
+  @Test
+  public void xtsTestsFromAwsLCPositive() {
+    final Stream<AesXtsTestCase> tests =
+        Stream.of(
+            new AesXtsTestCase()
+                .addKey("2718281828459045235360287471352662497757247093699959574966967627314159")
+                .addKey("2653589793238462643383279502884197169399375105820974944592")
```

Comment:
why use multiple builder calls for the key + tweak which are fixed size?

Reply:
It helps with breaking long lines. It's purely cosmetic :)